Sources/CollectionsExtension.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  CollectionsExtension.swift
+//  EZSwiftExtensions
+//
+//  Created by Sanyal, Arunav on 3/5/17.
+//  Copyright © 2017 Goktug Yilmaz. All rights reserved.
+//
+
+import Foundation
+
+extension Collection {
+    
+    // EZSE : A parralelized map for collections, operation is non blocking
+    public func pmap<R>(_ each: @escaping (Self.Iterator.Element) -> R) -> [R?] {
```

Comment:
My only callout here would be should the return type be [R?] or [R]. I can go either way. I am not sure as to the safety of these ops though so I kept it as [R?] (e.g. if each method returns a bunch of nils). However [R] is more consumer friendly.